viamrobotics/micro-rdk
# RSDK-4695 - installer CLI part 3: implement write and flash

This PR looks larger than it is. Unfortunately, for builds to work I have to commit the Cargo.lock file which creates a very large diff because of espflash's dependencies

Testing instructions

1. Download the viam.json config file from app
2. From the top level of the directory (not from within examples), run `MICRO_RDK_USE_NVS=true make build-esp32-with-cred-bin`
3. `cd micro-rdk-installer && cargo run -- write-flash --app-config=<path to viam.json file> --binary-path=<path to local copy of micro-rdk repo>/examples/target/esp32-server-with-cred.bin --monitor` 
4. Follow the instructions in the terminal (it should ask you to input your wi-fi SSID and then your wi-fi password)
